lib/services/presets.go

```diff
@@ -53,8 +53,11 @@ func NewPresetEditorRole() types.Role {
 					types.NewRule(types.KindOIDC, RW()),
 					types.NewRule(types.KindSAML, RW()),
 					types.NewRule(types.KindGithub, RW()),
+					types.NewRule(types.KindClusterAuditConfig, RW()),
 					types.NewRule(types.KindClusterAuthPreference, RW()),
-					types.NewRule(types.KindClusterConfig, RW()),
+					types.NewRule(types.KindClusterName, RW()),
+					types.NewRule(types.KindClusterNetworkingConfig, RW()),
+					types.NewRule(types.KindSessionRecordingConfig, RW()),
```

Comment:
Do you know if these preset roles are updated on each start? If they aren't, we probably should add a migration to update "editor" role to include these new rules when v8 Teleport starts.

Reply:
No, they aren't (recall that `CreateRole` stores the new value only if the key isn't already set in the datastore): https://github.com/gravitational/teleport/blob/adb7c6ea128f6b6c168aa8e30e9354e0cded74e5/lib/auth/init.go#L529-L544

However, extending a role's privileges automatically (whether by migrating or by updating the built-in roles during Auth server init) may constitute a security risk as it may violate an expectation/invariant in place when the role was being assigned. Once a role is in the wild, its privileges should never be extended without explicit confirmation.

Moreover, we may even need to work with the possibility the role of the same name has never been created by Teleport in the first place but has been entirely specified by a cluster admin, in which case we must also strive to avoid any privilege update whatsoever.

Reply:
So what would happen to users who have the old `editor` role after upgrade to v8, would any functionality be affected? Just trying to ensure we're not breaking compatibility. If we can't migrate the role, would it make sense to at least check legacy `ClusterConfig` permission in the ACL layer to keep backwards compatibility?

Reply:
I don't think much functionality would be affected as the new smaller resources were already in v7 and the editor role was lacking the corresponding privileges back then as well.

However I like the ACL idea and implemented it in https://github.com/gravitational/teleport/pull/8150/commits/d88ea147ef54579c36102c900053e078f9d88018. What do you think?